File: MaWPictures/src/main/java/us/mikeandwan/photos/di/PreferenceModule.java
package us.mikeandwan.photos.di;

import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import us.mikeandwan.photos.prefs.CategoryDisplayPreference;
import us.mikeandwan.photos.prefs.NotificationPreference;
import us.mikeandwan.photos.prefs.PhotoDisplayPreference;
import us.mikeandwan.photos.prefs.SyncPreference;


@Module
public class PreferenceModule {
    @Provides
    @Singleton
    public CategoryDisplayPreference provideCategoryDisplayPreference(SharedPreferences sharedPrefs) {
        return new CategoryDisplayPreference(sharedPrefs);
    }


    @Provides
    @Singleton
    public NotificationPreference provideNotificationPreference(SharedPreferences sharedPrefs) {
        return new NotificationPreference(sharedPrefs);
    }


    @Provides
    @Singleton
    public PhotoDisplayPreference providePhotoDisplayPreference(SharedPreferences sharedPrefs) {
        return new PhotoDisplayPreference(sharedPrefs);
    }


    @Provides
    @Singleton
    public SyncPreference provideSyncPreference(SharedPreferences sharedPrefs) {
        return new SyncPreference(sharedPrefs);
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/di/PreferenceModule.kt
package us.mikeandwan.photos.di

import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import us.mikeandwan.photos.preferences.CategoryDisplayPreference
import us.mikeandwan.photos.preferences.NotificationPreference
import us.mikeandwan.photos.preferences.PhotoDisplayPreference
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
class PreferenceModule {
    @Provides
    @Singleton
    fun provideCategoryDisplayPreference(sharedPrefs: SharedPreferences): CategoryDisplayPreference {
        return CategoryDisplayPreference(sharedPrefs)
    }

    @Provides
    @Singleton
    fun provideNotificationPreference(sharedPrefs: SharedPreferences): NotificationPreference {
        return NotificationPreference(sharedPrefs)
    }

    @Provides
    @Singleton
    fun providePhotoDisplayPreference(sharedPrefs: SharedPreferences): PhotoDisplayPreference {
        return PhotoDisplayPreference(sharedPrefs)
    }
}